@here/harp-mapview/lib/MapView.ts

```diff
@@ -1752,6 +1757,39 @@ export class MapView extends THREE.EventDispatcher {
         return this.m_fog;
     }
 
+    /**
+     * Sets the DataSource which contains the elevations, the elevation range source, and the
+     * elevation provider. Only a single elevation source is possible per [[MapView]]
+     *
+     * If the terrain-datasource is merged with this repository, we could internally construct
+     * the [[ElevationRangeSource]] and the [[ElevationProvider]] and access would be granted to
+     * the application when it asks for it, to simplify the API.
+     *
+     * @param elevationSource The datasource containing the terrain tiles.
+     * @param elevationRangeSource Allows access to the elevation min / max per tile.
+     * @param elevationProvider Allows access to the elevation at a given location or a ray
+     *      from the camera.
+     */
+    setElevationSource(
+        elevationSource: DataSource,
+        elevationRangeSource: ElevationRangeSource,
+        elevationProvider: ElevationProvider
+    ) {
+        // Try to remove incase this method was already called, will do nothing if it doesn't exist.
+        this.removeDataSource(elevationSource);
```

Comment:
But this will invalidate cache etc ... Better check if it's already added.
Or make `addDataSource` idempotent (better idea IMHO).

Reply:
Maybe I will wait for: https://github.com/heremaps/harp.gl/pull/424/ ?